### PR TITLE
new option to add a secondary cluster

### PR DESCRIPTION
**Pull request by briskt**

### CODE REVIEW

File: terraform/000-core/main.tf [hunk 1]
```diff
@@ -16,9 +15,8 @@ module "ecscluster" {
 module "ecscluster_secondary" {
   count = var.aws_region_secondary == "" ? 0 : 1
 
-  source   = "github.com/silinternational/terraform-modules//aws/ecs/cluster?ref=8.0.1"
-  app_name = var.app_name
-  app_env  = "secondary"
+  source       = "github.com/silinternational/terraform-modules//aws/ecs/cluster?ref=8.1.0"
+  cluster_name = "idp-secondary"
```
Comment (Baggerone) [inline terraform/000-core/main.tf:19]: Do we really want this hard coded, one-size-fits-all?
Maybe "${var.app_name}-${var.app_env}-secondary" ?

Comment (Baggerone) [reply terraform/000-core/main.tf:19]: Or just "${var.app_name}-secondary"

**Review by Baggerone — APPROVED** [review]
Assuming you change the cluster_name or have a good reason to leave it hard coded. :-)

File: terraform/000-core/main.tf (hunk 1 above)
Comment (briskt) [reply terraform/000-core/main.tf]: Yeah, I don't know what I was thinking!

**Comment by Baggerone** [comment]
I guess I do have a question about this.
What's your plan for deciding whether these changes actually need to be separated out into a new workspace?

**Comment by briskt** [comment]
> I guess I do have a question about this. What's your plan for deciding whether these changes actually need to be separated out into a new workspace?

Great question. For this module, I made the decision based on whether or not it would ever need to be applied during a recovery scenario. Since everything in this module is created and ready to use, nothing will need to be done while AWS operations are unstable.

For modules like 010-cluster that won't be fully operational until failover time, we need to consider the possibility that a separate workspace might be needed. I say "possibility" because we may decide it's safe to keep everything in one workspace as long as we believe that the AWS control plane will continue to operate normally during an outage in our primary region. That necessarily means that we move our primary region away from us-east-1 into a different region (e.g. us-east-2) since most services have their control plane in us-east-1. (That is probably a good plan regardless of the decision about workspaces.)

File: terraform/000-core/README.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
 # 000-core - Core setup: IAM users, ECS cluster
-This module is used to create an ECS cluster along with the necessary
+This module is used to create one or two ECS clusters along with the necessary
 IAM roles to function. It can also optionally create an ACM certificate
 used by later workspaces for HTTPS with the ALB.
 
 ## What this does
 
  - Create ECS cluster named after `app_name` and `app_env`
+ - Optionally create a secondary ECS cluster named `app_name_-app_env-secondary`
```
[the remark this answers is not in the document]
Comment (briskt) [reply terraform/000-core/README.md:9]: Yes, thanks.

**Comment by briskt** [comment]
it's looking like Plan B is going to work better

**Pull request closed without merging**